tools/LogCollector/source/update_mgmt_health_check.py

```diff
@@ -0,0 +1,268 @@
+            directories.extend(dirnames)
+            break # Get the top level of directories
+
+        for directory in directories:
+            if len(directory) >= 32:
+                potential_workspaces.append(directory)
+
+        if len(potential_workspaces) > 1:
+            agent_status.append("OMS Agent may be multihomed. Potential workspaces: " + str(potential_workspaces) + "\n")
+
+    else:
+        agent_status.append("Directory /var/opt/microsoft/omsagent/ does not exist. Is the omsagent installed? \n")
+
+    return "".join(agent_status)
+
+def check_hybrid_worker_running():
+    return grep_for_process("worker", 
+                            ["python /opt/microsoft/omsconfig/modules/nxOMSAutomationWorker/DSCResources/MSFT_nxOMSAutomationWorkerResource/automationworker/worker/hybridworker.py"], 
+                            "Hybrid worker")
+
+def check_hybrid_worker_installed():
+    workspace = get_workspace()
+
+    if workspace is not None:
+        worker_conf_path = "/var/opt/microsoft/omsagent/" + workspace + "/state/automationworker/worker.conf"
+
+        if os.path.isfile(worker_conf_path):
+            return "worker.conf exists. Hybrid worker looks to be installed. \n"
+        else:
+            return "worker.conf does not exist. Is hybrid worker installed? \n"
+    else:
+        return "Checking hybrid worker installed...workspace not found. Is OMS Agent installed?"
+
+def check_oms_agent_running():
+    return grep_for_process("omsagent", ["omsagent.log", "omsagent.conf"], "OMS Agent")
+
+def grep_for_process(process_name, search_criteria, output_name):
+    grep_output = os.popen("ps aux | grep " + process_name).read()
+
+    output = []
+
+    if any(search_text in grep_output for search_text in search_criteria):
+        output.append(output_name + " seems to be running. \n\n")
+    else:
+        output.append(output_name + " does not seem to be running. \n\n")
+    
+    output.append(output_name + " output: \n" + str(grep_output))
+
+    return "".join(output)
+
+def check_oms_admin():
+    oms_admin_path = "/etc/opt/microsoft/omsagent/conf/omsadmin.conf"
+
+    output = []
+
+    output.append("OMS Admin Conf: \n\n")
+
+    if os.path.isfile(oms_admin_path):
+        oms_admin_file = open(oms_admin_path, 'r')
+
+        for line in oms_admin_file:
+            output.append(line)
+        return "".join(output)
+    else:
+        return "omsadmin.conf file not found"
+
+def get_workspace():
+    oms_admin_path = "/etc/opt/microsoft/omsagent/conf/omsadmin.conf"
+    line = find_line_in_path("WORKSPACE", oms_admin_path)
+
+    if line is not None:
+        return line.split("=")[1].strip()
+
+    return None
+
+def get_machine_info():
+    output = []
+
+    output.append("\nMachine Information: \n")
+    hostname_output = os.popen("hostnamectl").read()
+    output.append(hostname_output + "\n")
+
+    return "".join(output)
+
+def check_ff():
```

Comment:
how about mooncake?

Reply:
update management currently is not supported in mooncake, so I omitted it.